include/spl/sys/vmem.h

```diff
@@ -98,10 +98,14 @@ extern size_t vmem_size(vmem_t *vmp, int typemask);
 #define	vmem_zalloc(sz, fl)	spl_vmem_zalloc((sz), (fl), __func__, __LINE__)
 #define	vmem_free(ptr, sz)	spl_vmem_free((ptr), (sz))
 #define	vmem_qcache_reap(ptr)	((void)0)
+#define	kvmem_alloc(sz, fl)	spl_kvmem_alloc((sz), (fl), __func__, __LINE__)
+#define	kvmem_zalloc(sz, fl)	spl_kvmem_zalloc((sz), (fl), __func__, __LINE__)
```

Comment:
Rather than introducing a new interface, have we considered unconditionally using `kvmalloc` for `vmalloc` when it's available.  With the possible exception of the SPL slabs, I'm not aware of any `vmem_alloc()` callers which wouldn't benefit from this.  This would simplify the patch and help avoid any portability concerns.

Reply:
using kvmalloc unconditionally for any vmem allocation can be considered, but its not my part to make this decision. so i implemented it conditionally for zstd compression and other workspace allocations like gzip. since the kernel does offer it as conditional allocator i decided to export it in that way.  if you wish a unconditional use for any of the vmem allocation types i'm willing todo so, right now its already unconditionally used if size <= spl_kmem_alloc_max to follow the guide given by zfs maintainers not to use physical kmem on bigger sizes than specified by this variable.

Reply:
@behlendorf why should SPL slabs not use kvmem? (the current patch does this: spl_slab_alloc->kv_alloc)

Reply:
> using kvmalloc unconditionally for any vmem allocation can be considered, but its not my part to make this decision.

Understood, that makes sense and I appreciate it!  My specific concern here is that adding these new interfaces will result in additional work and compatibility code for the other OpenZFS platforms.   I'm trying to suss out if we have a real need for them, and I don't think we do.  It seems like every existing `vmem_alloc()` caller should potentially benefit from using `kvmalloc()` (when it's available).  Are there any places where we would not want this?

> "With the possible exception of the SPL slabs".

I should have included the word "large" in there.  What I was trying to get at is that we do have a couple places (like slabs with large objects, or the workspaces) were we do know upfront based on the `size` that the allocation cannot be satisfied by a `kmalloc()`.  In which case, why try.  That said, it looks like the kernel already does a nice job checking for and doing something smart.  Worst case if we find we need it we could always add a flag as a hint, to only try the `vmalloc`.